pkg/virtctl/portforward/portforward.go

```diff
@@ -212,66 +219,68 @@ func examples() string {
 }
 
-// ParseTarget argument supporting the form of vmi/name.namespace (or simpler)
-func ParseTarget(target string) (string, string, string, error) {
-	kind := "vmi"
+// ParseTarget argument supporting the form of type/name[/namespace] or the legacy form of type/name.namespace
+func ParseTarget(target string) (kind string, namespace string, name string, err error) {
+	if target == "" {
+		return "", "", "", errors.New("target cannot be empty")
+	}
 
 	parts := strings.Split(target, "/")
-	if len(parts) > 2 {
-		return "", "", "", errors.New("target is not valid with more than one '/'")
-	}
-	if len(parts) == 2 {
+	switch len(parts) {
+	case 1:
+		return "", "", "", errors.New("target must contain type and name separated by '/'")
+	case 2:
+		kind = parts[0]
+		name = parts[1]
+	case 3:
 		kind = parts[0]
-		if !kindIsVM(kind) && !kindIsVMI(kind) {
-			return "", "", "", errors.New("unsupported resource kind " + kind)
+		name = parts[1]
+		namespace = parts[2]
+		if namespace == "" {
+			return "", "", "", errors.New("namespace cannot be empty")
 		}
-		target = parts[1]
-	} else {
-		log.Log.Warningf("Defaulting to type '%s'.\nOmitting the type in the target is deprecated."+
-			"\nSpecifying the type will become mandatory with the next release.", kind)
+	default:
+		return "", "", "", errors.New("target is not valid with more than two '/'")
 	}
-	if target == "" {
-		return "", "", "", errors.New("expected name after '/'")
-	}
-	if target[0] == '.' {
-		return "", "", "", errors.New("expected name before '.'")
+
+	kind, err = normalizeKind(kind)
+	if err != nil {
+		return "", "", "", err
 	}
-	if target[len(target)-1] == '.' {
-		return "", "", "", errors.New("expected namespace after '.'")
+
+	if name == "" {
+		return "", "", "", errors.New("name cannot be empty")
 	}
 
-	name := target
-	namespace := ""
-	if lastDot := strings.LastIndex(target, "."); lastDot != -1 {
-		name = target[:lastDot]
-		namespace = target[lastDot+1:]
+	if lastDot := strings.LastIndex(parts[1], "."); len(parts) == 2 && lastDot > -1 {
+		log.Log.Warning("A dot in target without specifying a namespace activates legacy parsing of name and namespace.\n" +
+			"This behavior will be removed in the next release.")
+		name = parts[1][:lastDot]
+		namespace = parts[1][lastDot+1:]
 	}
 
 	return kind, namespace, name, nil
 }
-func kindIsVMI(kind string) bool {
-	return kind == "vmi" ||
-		kind == "vmis" ||
-		kind == "virtualmachineinstance" ||
-		kind == "virtualmachineinstances"
-}
 
-func kindIsVM(kind string) bool {
-	return kind == "vm" ||
-		kind == "vms" ||
-		kind == "virtualmachine" ||
-		kind == "virtualmachines"
+func normalizeKind(kind string) (string, error) {
+	switch kind {
+	case vm, "vms", "virtualmachine", "virtualmachines":
+		return vm, nil
+	case vmi, "vmis", "virtualmachineinstance", "virtualmachineinstances":
+		return vmi, nil
+	}
+	return "", fmt.Errorf("unsupported resource type '%s'", kind)
 }
```

Comment:
Would be a good idea to lower case the `kind` string before the switch case?

Reply:
Good idea! Updated